goal_src/engine/game/settings.gc

```diff
@@ -5,5 +5,734 @@
+    (set! (-> gp-0 ambient) (-> s5-0 ambient))
+    (set! (-> gp-0 allow-pause) (-> s5-0 allow-pause))
+    (set! (-> gp-0 allow-progress) (-> s5-0 allow-progress))
+    (set! (-> gp-0 allow-look-around) (-> s5-0 allow-look-around))
+    (set! (-> gp-0 ocean-off) (-> s5-0 ocean-off))
+    (set! (-> gp-0 ambient-volume-movie) (-> s5-0 ambient-volume-movie))
+    (set! (-> gp-0 music-volume-movie) (-> s5-0 music-volume-movie))
+    (set! (-> gp-0 sfx-volume-movie) (-> s5-0 sfx-volume-movie))
+    (set! (-> gp-0 dialog-volume-hint) (-> s5-0 dialog-volume-hint))
+    (set! (-> gp-0 process-mask) (-> s5-0 process-mask))
+    )
+   (set!
+    (-> *kernel-context* prevent-from-run)
+    (the-as process-mask (-> gp-0 process-mask))
+    )
+   gp-0
+   )
+  )
+
+;; definition for method 13 of type setting-control
+(defmethod TODO-RENAME-13 setting-control ((obj setting-control))
+  (TODO-RENAME-12 obj)
+  (let ((gp-0 (-> obj current)))
+   (let ((s5-1 (-> obj target)))
+    (when *sound-player-enable*
+     (when (!= (-> gp-0 sfx-volume) (-> s5-1 sfx-volume))
+      (set!
+       (-> gp-0 sfx-volume)
+       (seek
+        (-> gp-0 sfx-volume)
+        (-> s5-1 sfx-volume)
+        (* 100.0 (-> *display* seconds-per-frame))
+        )
+       )
+      (sound-set-volume (the-as uint 1) (-> gp-0 sfx-volume))
+      )
+     (when (!= (-> gp-0 music-volume) (-> s5-1 music-volume))
+      (set!
+       (-> gp-0 music-volume)
+       (seek
+        (-> gp-0 music-volume)
+        (-> s5-1 music-volume)
+        (* 100.0 (-> *display* seconds-per-frame))
+        )
+       )
+      (sound-set-volume (the-as uint 2) (-> gp-0 music-volume))
+      )
+     (when (!= (-> gp-0 dialog-volume) (-> s5-1 dialog-volume))
+      (set!
+       (-> gp-0 dialog-volume)
+       (seek
+        (-> gp-0 dialog-volume)
+        (-> s5-1 dialog-volume)
+        (* 100.0 (-> *display* seconds-per-frame))
+        )
+       )
+      (sound-set-volume (the-as uint 4) (-> gp-0 dialog-volume))
+      )
+     (when (!= (-> gp-0 ambient-volume) (-> s5-1 ambient-volume))
+      (set!
+       (-> gp-0 ambient-volume)
+       (seek
+        (-> gp-0 ambient-volume)
+        (-> s5-1 ambient-volume)
+        (* 100.0 (-> *display* seconds-per-frame))
+        )
+       )
+      (sound-set-volume (the-as uint 16) (-> gp-0 ambient-volume))
+      )
+     )
+    (when (!= (-> gp-0 language) (-> s5-1 language))
+     (set! (-> gp-0 language) (-> s5-1 language))
+     (set-language (-> gp-0 language))
+     )
+    (when
+     (and
+      (!= (-> s5-1 music) (-> gp-0 music))
+      (and
+       (< 0.0 (-> *setting-control* current music-volume))
+       (zero? (rpc-busy? 1))
+       *sound-bank-1*
+       *sound-bank-2*
+       )
+      )
+     (cond
+      ((-> s5-1 music)
+       (format 0 "Load music ~A~%" (-> s5-1 music))
+       (sound-music-load
+        (string->sound-name
+         (the-as
+          string
+          (-> (the-as (pointer uint32) (+ #xff38 (the-as int (-> s5-1 music)))))
```

Comment:
something weird happened here. Do you have a cast here?  This should be a symbol->string conversion.